### DISCUSSION TITLE
Ubuntu 20.04.6 LTS pycodec2 fail

### DISCUSSION THREAD

**Discussion by ashworthpayne** [markqvist/Reticulum#555] [Help & Questions] [answered]
Building wheels for collected packages: pycodec2
  Building wheel for pycodec2 (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for pycodec2 (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      running bdist_wheel
      running build
      running build_py
      running build_ext
      building 'pycodec2' extension
      error: unknown file type '.pxd' (from 'pycodec2/codec2.pxd')
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pycodec2
Failed to build pycodec2
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (pycodec2)

I'm not well-versed in these sorts of problems so any help would be appreciated.

**Comment by markqvist** [comment]
Moving this to the discussion forum for greater visibility.

**Comment by markqvist** [comment]
I can confirm this error on occurs on a fresh installation of Ubuntu 20.04. This is a problem with the install/build process for the `pycodec2` package, which does not support installing on Ubuntu 20.04. Unfortunately I can't do much about this directly, since it requires updates to the `pycodec2` package that I do not maintain.

Most realistic path to resolve this is to upgrade to Ubuntu 22.04, which supports installing just fine via `pipx`. Alternatively, you can ask the developer of `pycodec2` if he's willing to add support for Ubuntu 20.04: https://github.com/gregorias/pycodec2

I've updated the install guide to make it more clear that `pipx` / `pip` installation is only expected to work on Ubuntu 22.04+ and Debian 11+.

Thanks for noticing it!

**Comment by ashworthpayne** [comment, marked as answer]
I upgraded to 22.04 and the install completed as expected. Thank you for taking the time to explain the cause of the error.
